### ISSUE TITLE
ion-select should re-evaluate .text when list options updated

### ISSUE THREAD

**Issue by webprofusion-chrisc** [ionic-team/ionic#15788]
# Bug Report

```
Ionic:

   ionic (Ionic CLI)          : 4.1.2 (/usr/local/lib/node_modules/ionic)
   Ionic Framework            : @ionic/angular 4.0.0-beta.12
   @angular-devkit/core       : 0.7.2
   @angular-devkit/schematics : 0.7.2
   @angular/cli               : 6.1.5
   @ionic/ng-toolkit          : 1.0.7
   @ionic/schematics-angular  : 1.0.5

Cordova:

   cordova (Cordova CLI) : 8.0.0
   Cordova Platforms     : android 7.1.1, browser 5.0.4, ios 4.5.5
   Cordova Plugins       : cordova-plugin-ionic-keyboard 2.1.2, cordova-plugin-ionic-webview 2.0.2, (and 7 other plugins)

System:

   Android SDK Tools : 26.1.1 (/Users/chrisc/Library/Android/sdk)
   ios-deploy        : 2.0.0
   ios-sim           : 7.0.0
   NodeJS            : v8.11.3 (/usr/local/bin/node)
   npm               : 6.4.1
   OS                : macOS High Sierra
   Xcode             : Xcode 10.0 Build version 10A255
```

**Describe the Bug**
If an ion-select has ion-select-options (ngFor) bound to an array which is then updated async, the select options update OK but the selected value shown in the UI is not. This is because ion-select only sets this.text when this.valueChanged fires, not when the related ion-select-options children are refreshed. 

https://github.com/ionic-team/ionic/blob/master/core/src/components/select/select.tsx#L123 

Workaround:
Binding [selectedText] to the required selected value works:
```
<ion-select [(ngModel)]="offer.growerNumber" name="growerNumber" okText="OK" cancelText="Dismiss" placeholder="Choose..." [selectedText]="offer.growerNumber">
            <ion-select-option *ngFor="let i of growerNumbers" [value]="i">{{i}}</ion-select-option>
          </ion-select>
```
**Expected Behavior**
When data source for ion-select-option list resolves via async call, the selected value should be shown in the UI.

**Additional Context**
https://forum.ionicframework.com/t/ionic-4-ion-select-and-change-detection/141643/2

**Comment by paulstelzer** [comment]
Thanks for your issue! Could you please check if this is still an issue in `beta.17`? That would be very nice :)

**Comment by denisroldan** [comment]
@paulstelzer Experiencing the same issue here 

```

Ionic:

   ionic (Ionic CLI)             : 4.5.0 (/usr/local/lib/node_modules/ionic)
   Ionic Framework               : @ionic/angular 4.0.0-beta.17
   @angular-devkit/build-angular : 0.10.7
   @angular-devkit/schematics    : 7.0.7
   @angular/cli                  : 7.0.7
   @ionic/angular-toolkit        : not installed

System:

   NodeJS : v10.11.0 (/usr/local/bin/node)
   npm    : 6.4.1
   OS     : macOS Mojave
```

**Comment by denisroldan** [comment]
@paulstelzer It magically works now. Sorry! 😅 From my side, it just works as expected on Ionic 4 latest ;)

**Comment by paulstelzer** [comment]
@webprofusion-chrisc Could you check if your issue is also solved when updating to beta.18?

**Comment by webprofusion-chrisc** [comment]
Hi, no not for me but perhaps my test case is wrong:

```html
     <ion-select [(ngModel)]="selectedthing" okText="OK" cancelText="Dismiss" placeholder="Choose...">
          <ion-select-option *ngFor="let i of mylist" [value]="i">{{i}}</ion-select-option>
        </ion-select>
        
        {{status}}
```

```typescript

```

**Comment by sneko** [comment]
I think I have the exact same issue. I use something like:
```
              ion-select(... :selectedText="element.id")
                ion-select-option(v-for='element in elements' :value='element.id') {{ element.id + ' ' + element.name }}
```

Is it still an issue on your side?

_Note that it was working with an old Angular+Ionic2 project_
_Note2: the syntax is due to Vue.js ;)_

**Comment by nomanshahsyed** [comment]
Hi, has anyone found the solution to this problem? I have the exact same problem. I have different pages and they share a same array. When some of the page POPs or SPLICEs an entry then the select options are not updated in the UI. Although they seem to be changed in the console.

Another tricky thing is that the select options are updated on one page but not on the other one. The difference in these pages is that the one where teh options are not updated, the ion-select is wrapped around a DIV.

**Comment by liamdebeasi** [comment]
A fix for this is available in Ionic 4.9.0. For any new bugs, please create a new issue.

Thanks!
